File: samples/conference/payments-bc/src/main/java/com/microsoft/conference/payments/domain/Events/PaymentInitiated.cs
using System;
using System.Collections.Generic;
using ENode.Eventing;

namespace Payments
{
    public class PaymentInitiated : DomainEvent<Guid>
    {
        public Guid OrderId { get; private set; }
        public Guid ConferenceId { get; private set; }
        public string Description { get; private set; }
        public decimal TotalAmount { get; private set; }
        public IEnumerable<PaymentItem> Items { get; private set; }

        public PaymentInitiated() { }
        public PaymentInitiated(Guid orderId, Guid conferenceId, string description, decimal totalAmount, IEnumerable<PaymentItem> items)
        {
            OrderId = orderId;
            ConferenceId = conferenceId;
            Description = description;
            TotalAmount = totalAmount;
            Items = items;
        }
    }
}

File: samples/conference/payments-bc/src/main/java/com/microsoft/conference/payments/domain/events/PaymentInitiated.java
package com.microsoft.conference.payments.domain.events;

import com.microsoft.conference.payments.domain.models.PaymentItem;
import lombok.Getter;
import lombok.Setter;
import org.enodeframework.eventing.DomainEvent;

import java.math.BigDecimal;
import java.util.List;

@Getter
@Setter
public class PaymentInitiated extends DomainEvent<String> {
    private String orderId;
    private String conferenceId;
    private String description;
    private BigDecimal totalAmount;
    private List<PaymentItem> paymentItems;

    public PaymentInitiated() {
    }

    public PaymentInitiated(String orderId, String conferenceId, String description, BigDecimal totalAmount, List<PaymentItem> paymentItems) {
        this.orderId = orderId;
        this.conferenceId = conferenceId;
        this.description = description;
        this.totalAmount = totalAmount;
        this.paymentItems = paymentItems;
    }
}
